internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer.go

```diff
@@ -40,6 +40,31 @@ func (p PhaseList) AppendWhen(when bool, name string, tasks ...runtime.TaskSetup
 	return p
 }
 
+// ApplyConfiguration is a sequence which is called to apply a new configuration to the system.
+// It will reboot subsequent make it active.
+func (*Sequencer) ApplyConfiguration(r runtime.Runtime, req *machineapi.ApplyConfigurationRequest) []runtime.Phase {
+	phases := PhaseList{}
+
+	phases = phases.Append(
+		"mountState",
+		MountStatePartition,
+	).Append(
+		"saveConfig",
+		SaveConfig,
+	).Append(
```

Comment:
I think we should do exactly what reboot sequence does once we are done saving the config:

```
		phases = phases.Append(
			"stopEverything",
			StopAllServices,
		).Append(
			"umount",
			UnmountOverlayFilesystems,
			UnmountPodMounts,
		).Append(
			"unmountSystem",
			UnmountEphemeralPartition,
			UnmountStatePartition,
		).Append(
			"unmountBind",
			UnmountSystemDiskBindMounts,
		).Append(
			"reboot",
			Reboot,
		)
```